### ISSUE TITLE
SNI Bypass possible?

### ISSUE THREAD

**Issue by PYUDNG** [tachiyomiorg/tachiyomi#7703]
### Describe your suggested feature

Some websites(or their web server / hostname) has been blocked by sni (e.g. exhentai pixiv etc.). Is is possible to let users configure a host-sni map(or a pre-set one) to make them available?
As I know, there is some github repo like EhViewe,  Pixez-Flutter have already implemented. Is it possible for tachiyomi as well? Thanks.

### Other details

https://github.com/Notsfsssf/pixez-flutter
https://gitlab.com/NekoInverter/EhViewer

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an extension, I should be opening an issue in the [extensions repository](https://github.com/tachiyomiorg/tachiyomi-extensions/issues/new/choose).
- [X] I have updated the app to version **[0.13.5](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I will fill out all of the requested information in this form.

**Comment by arkon** [comment]
I don't fully understand what [EhViewer is doing](https://gitlab.com/NekoInverter/EhViewer/-/blob/master/app/src/main/java/com/hippo/ehviewer/client/EhSSLSocketFactory.java#L30).

I'd probably just wait for OkHttp and Conscrypt to support ESNI though:

- https://github.com/square/okhttp/issues/6539
- https://github.com/google/conscrypt/issues/730

**Comment by PYUDNG** [comment]
@arkon I'm glad to get your reply! What EhViewer actually doing is [Domain_fronting](https://en.wikipedia.org/wiki/Domain_fronting) which aims to bypass ISP's hostname blacklist.
Some ISP(e.g. ISP of China, Korean, etc) inspects network content and get hostname from SNI, and blocks its connection if they found it in their blacklist. What SNI Bypassing trying to do is to remove or modify the SNI, to make their ISP unable to filter it out.

**Comment by yschimke** [comment]
Domain Fronting should already work with OkHttp. Added in https://github.com/repairnator/repairnator-experiments/commit/afeb62290d4ceac120b21d33ab9fc13e8b105a1d

ECH is probably a way off, blocked on a SSL provider that supports it.

**Comment by stevenyomi** [comment]
Domain fronting can be done on the extension side by building its own version of OkHttp client. Open an issue in the [extensions repository](https://github.com/tachiyomiorg/tachiyomi-extensions/issues/new/choose) for a specific extension if needed.

**Comment by Solovertical** [comment]
> Domain fronting can be done on the extension side by building its own version of OkHttp client. Open an issue in the [extensions repository](https://github.com/tachiyomiorg/tachiyomi-extensions/issues/new/choose) for a specific extension if needed.

Recently on some specific Internet provider you face issue like  you can't  download extension without VPN  but on other providers you can is it some kind of SNI blocked and also is it solved the ISP blacklist like this. You can visit this on other providers but not which mentioned above

![Screenshot_2023-04-09-23-56-59-634_xyz jmir tachiyomi mi debug](https://user-images.githubusercontent.com/109901807/230790527-d5d57539-d1ec-4080-bc2b-71872af7e025.jpg)

**Comment by Solovertical** [comment]
> Domain fronting can be done on the extension side by building its own version of OkHttp client. Open an issue in the [extensions repository](https://github.com/tachiyomiorg/tachiyomi-extensions/issues/new/choose) for a specific extension if needed.

Recently on some specific Internet provider you face issue like you can't download extension without VPN but on other providers you can, Is it Some Some kind of SNI blocked ,And also is domain Fronting can solve the ISP blacklist like this ,You can access this on other providers but not on which Mentioned above 

![Screenshot_2023-04-09-23-56-59-634_xyz jmir tachiyomi mi debug](https://user-images.githubusercontent.com/109901807/230791079-63985168-d426-4720-be42-8a34fe1bf258.jpg)

**Comment by stevenyomi** [comment]
@Solovertical This is not Aniyomi repo.
